src/wallet/rpcwallet.cpp

```diff
@@ -1836,8 +1933,15 @@ UniValue listtransactions(const JSONRPCRequest& request)
     pwallet->BlockUntilSyncedToCurrentChain();
 
     std::string strAccount = "*";
-    if (!request.params[0].isNull())
-        strAccount = request.params[0].get_str();
+    if (!IsDeprecatedRPCEnabled("accounts")) {
+        if (request.params[0].get_str() != "*") {
```

Comment:
If `params[0]` is unset, this will throw an error at `get_str()` won't it?
Looks like you want
```C++
if (!request.params[0].isNull()) {
  strAccount = request.params[0].get_str();
  if (!IsDeprecatedRPCEnabled("accounts") && strAccount != "*") {
    throw ...
  }
}
```

Reply:
Correct, `accounts` is optional.

Reply:
Yes, much better. Thanks